### ISSUE TITLE
Remove Cull Less Leaves

### ISSUE THREAD

**Issue by Madis0** [Fabulously-Optimized/fabulously-optimized#424]
### CurseForge link

https://www.curseforge.com/minecraft/mc-mods/cull-less-leaves

### CurseForge Mod Distribution

Allowed

### Modrinth link

https://modrinth.com/mod/cull-less-leaves

### Source/other link

https://github.com/isXander/CullLessLeaves

### Mod file size

42.63 KB

### What it does

Makes leaves faster without compromising on appearance 

### Why should it be in the modpack

The implementation might be better-looking and more performant than MoreCulling

### Why shouldn't it be in the modpack

MoreCulling seems to obsolete this, while still having fast updates and Sodium integration. 

### Categories

Performance optimization

### Additional details

If obsoleted, the best config for MoreCulling should be decided (it has 3 methods of implementing leaf culling)

**Please test https://github.com/Fabulously-Optimized/fabulously-optimized/issues/424#issuecomment-1204009338**

**Comment by Julienraptor01** [comment]
Pure Performance comparison between CullLessLeaves on 2, MoreCulling Depth 2 and Nothing
The point here is to see if CLL and MC use the same options which one is better and Nothing is just here to see if actually these mods are better than disabled

Instruction (Part1) :
Download a fresh 1.19.1 instance of FO. (Don't add any mods)
You want to do this to have a clean default config so that we're all on the same base.
Go to Video settings in the General tab and disable Vsync then go in the Quality tab and disable clouds then go in the Extras tab and enable Show FPS and show coordinates.
Hit apply.
###### alternative 1
Go then create a world in single player.
Change the game mode to creative, go to world updates in gamerules and disable advance time of day and put this world seed : `-3678181204597126149`.
Generate the world and once you're in do `/time set 6000`
###### alternative 2
Download this zipped world folder, unzip it and put it in your saves folder inside of your .minecraft. (the location of the .minecraft vary with your launcher)
###### end of alternatives
Once you've done that use this command `/execute in minecraft:overworld run tp @s 57.48 237.36 -391.96 -65.25 90.00`.
###### Special Marker start (because you'll be doing the same steps)
Now turn around to be sure every chunk is generated and loaded correctly and use the command again.
Wait a bit  for FPS to stabilize and screenshot with F2.
Now go to Video settings in Performance tab and disable CullLessLeaves then go in the MoreCulling tab, enable Include Mangrove Roots and change leaves culling to Depth.
Apply go back to game rerun the command to be sure you're in same place, wait for the FPS to stabilize and do a screenshot.
Now for the last step go back to the video settings in moreculling tab and switch back leaves culling to default and disable include mangrove roots.
Apply go back in game wait for the FPS to stabilize and screenshot.
Then go to Video settings in Performance tab and disable CullLessLeaves
###### Special Marker end (because you'll be doing the same steps)

Instruction (Part2) :
Do `/execute in minecraft:overworld run tp @s 72.23 110.00 -349.07 3.75 33.60`
Repeat from Special Marker Start to Special Marker End 

Instruction (Part3) :
Do `/execute in minecraft:overworld run tp @s -1259.13 231.62 1059.06 197.40 90.00`
Repeat from Special Marker Start to Special Marker End 

Instruction (Part4) :
Do `/execute in minecraft:overworld run tp @s -1339.13 119.00 1017.45 -287.85 46.35`
Repeat from Special Marker Start to Special Marker End 

Instruction (Part5) :
Do `/execute in minecraft:overworld run tp @s 2353.41 205.01 -746.58 213.76 90.00`
Load chunks and wait like 5mins to make sure leaves are stabilized.
This is due do an issue in swamps where leaves are too far from trunk and start despawning and cause lag.
We want here lag to be gone so that testing is coherent.
Repeat from Special Marker Start to Special Marker End 

Instruction (Part6) :
Do `/execute in minecraft:overworld run tp @s 2289.05 87.00 -751.00 -71.54 35.70`
Repeat from Special Marker Start to Special Marker End 

We're nearly done !!!
Our last test will be a jungle

Instruction (Part7) :
Do `/execute in minecraft:overworld run tp @s 2873.03 222.13 -2355.33 -750.79 90.00`
Repeat from Special Marker Start to Special Marker End

Yep after that you'll be done in Minedract !
Instruction (Part8) :
Do `/execute in minecraft:overworld run tp @s 2890.70 139.00 -2311.30 -937.09 49.35`
Repeat from Special Marker Start to Special Marker End

Nice you're finally done !
Now it's time to upload all these screenshots to imgur.
Put them a description pls otherwise it's useless
Your pics should be in this order with these names :

1. CLL Forest Fly
2. MC Forest Fly
3. Nothing Forest Fly
4. CLL Forest Ground
5. MC Forest Ground
6. Nothing Forest Ground
7. CLL Dark Forest Fly
8. MC Dark Forest Fly
9. Nothing Dark Forest Fly
10. CLL Dark Forest Ground
11. MC Dark Forest Ground
12. Nothing Dark Forest Ground
13. CLL Mangrove Fly
14. MC Mangrove Fly
15. Nothing Mangrove Fly
16. CLL Mangrove Ground
17. MC Mangrove Ground
18. Nothing Mangrove Ground
19. CLL Jungle Fly
20. MC Jungle Fly
21. Nothing Jungle Fly
22. CLL Jungle Ground
23. MC Jungle Ground
24. Nothing Jungle Ground

Now put as name (replace $username by your username) : CLL 2 vs MC Depth 2 vs Nothing ($username)
Grab link and paste it here with your specs.
It should look like that

# Example (actually it's my test)
https://imgur.com/a/xiCEAFv
MSI GE75 Raider 9SE
Intel(R) Core(TM) i7-9750H
NVIDIA GeForce RTX 2060 Mobile
Windows 10

**Comment by Julienraptor01** [comment]
Easier guide by @wafflecoffee !
https://gist.github.com/wafflecoffee/319973048923c0341ff476ae8948c725

**Comment by Julienraptor01** [comment]
My result :
https://imgur.com/a/xiCEAFv
MSI GE75 Raider 9SE
Intel(R) Core(TM) i7-9750H
NVIDIA GeForce RTX 2060 Mobile
Windows 10
MC seems to be better than Vanilla which seems to be slightly better than CLL
and in addition to that MC seem closer in look to Vanilla while CLL is closer to OptiFine fast leaves

**Comment by Julienraptor01** [comment]
> My result: https://imgur.com/a/O306Fh9
> Changed Configs: Render distance from 12 to 9
> Graphics Card: Intel(R) HD Graphics 610 (Integrated Graphics)
> CPU: Intel(R) Pentium(R) CPU 4417U
> Memory: 4GB

Were you on quake pro or smth ? There's a lot of distortion

**Comment by isXander** [comment]
I'd like to mention that CLL performance heavily depends on Sodium. So make sure to test with FO, not vanilla.

**Comment by Julienraptor01** [comment]
> I'd like to mention that CLL performance heavily depends on Sodium. So make sure to test with FO, not vanilla.

Yeah we know and don't test this with vanilla, we test this on FO

**Comment by Julienraptor01** [comment]
Updated gist
https://gist.github.com/Julienraptor01/8404861d7079222f35ba0646b35cdfdc
Made simple by not have to switch settings every screenshot.

**Comment by osfanbuff63** [comment]
Test results:

OS: Windows 10

https://www.dropbox.com/s/8t7l46kus1bgu5l/CLL%20MoreCulling%20Tests%20-%20osfanbuff63.zip?dl=0

**Comment by Madis0** [comment]
I'll make a decision based on your tests on Sunday.

**Comment by wati2006** [comment]
My results:
https://www.dropbox.com/s/jmxbxhsg9584otb/cll-mc-vanilla%20test.zip?dl=0
CPU: Intel(R) Core(TM) i5-7200U
GPU: NVIDIA GeForce 940MX

**Comment by Madis0** [comment]
> I'll make a decision based on your tests on Sunday.

Unfortunately I was not able to draw a conclusion yet. Needs more (and maybe different?) testing.

**Comment by Madis0** [comment]
MoreCulling inclusion was reverted, therefore this is postponed.

**Comment by Julienraptor01** [comment]
i restested moreculling and cull less leaves and vanilla on hypixel skyblock foraging island
moreculling is still superior to cull less leaves, both on the same mode (depth 2 which the look is different from vanilla)
and also on a moreculling specific mode (check surroundings which look like vanilla and still seem as perfomant than CLL depth 2)
+ moreculling has culling options that cull less leaves had so from my opinion it should at least be added whitout the leaves culling options
i said at least because i think it should replace CLL
TL;DR
For FPS :
MoreCulling Depth 2 > MoreCulling Check Surroundings > Cull Less Leaves Depth 2 > Vanilla
For vanilla look :
MoreCulling Check Surroundings > Vanilla (same look but less perfomance) > MoreCulling / Cull Less Leaves Depth 2 (not vanilla look-a-like so rejected)
